### PR TITLE
[BUG] docs that miss the sortby field may aappear before docs that has it

### LINKED ISSUE #2978: Sortby on missing fields gives priority to null fields
For example,
```
FT.CREATE idx SCHEMA t TEXT
HSET doc1 t 1
HSET doc2 somethingelse 2
FT.SEARCH idx * SORTBY t
```
Result: `[2, 'doc2', ['somethingelse', '2'], 'doc1', ['t', '1']]`

### PR DESCRIPTION
**Pull request by ashtul**
Currently, if a document is missing the sortby field, after the heap compare function assigned to it `-1` , it might change the sign to `1` if the sortby order is `ascending`. This caused results from fields from a missing value to come first while we would like them to arrive at the end.

Requires a fix for the coordinator as well.

Fix #2978

### CODE REVIEW

File: coord/src/module.c [hunk 1]
```diff
@@ -662,14 +662,15 @@ static int cmp_results(const void *p1, const void *p2, const void *udata) {
...
-      cmp = r2->sortKey ? 1 : -1;
+      return r2->sortKey ? 1 : -1;
     }
     // in case of a tie - compare ids
     if (!cmp) {
       // printf("It's a tie! Comparing <N=%lu> %.*s vs <N=%lu> %.*s\n", r2->idLen, (int)r2->idLen,
       //        r2->id, r1->idLen, (int)r1->idLen, r1->id);
       cmp = cmpStrings(r2->id, r2->idLen, r1->id, r1->idLen);
     }
+    //
```
Comment (MeirShpilraien) [inline coord/src/module.c:673]: Remove?

File: coord/src/module.c [hunk 2]
```diff
@@ -750,6 +751,8 @@ searchResult *newResult(searchResult *cached, MRReply *arr, int j, searchReplyOf
       if (eptr != res->sortKey + 1 && *eptr == 0) {
         res->sortKeyNum = d;
       }
+    } else if (!strncmp(res->sortKey, "none", 4)) {
+      res->sortKey = NULL;
```
Comment (MeirShpilraien) [inline coord/src/module.c:755]: Do we need to free `res->sortKey`?

**Review by MeirShpilraien — COMMENTED** [review]
:+1: 
Small comments

File: coord/src/module.c (hunk 2 above)
Comment (ashtul) [reply coord/src/module.c:755]: No. We do not take ownership of the MRReply elements.

Comment (MeirShpilraien) [reply coord/src/module.c:755]: So who will free it?

Comment (ashtul) [reply coord/src/module.c:755]: `MRCtx_Free` calls `MRReply_Free` on all the replies.

Comment (MeirShpilraien) [reply coord/src/module.c:755]: Did you run it with valgrind? its on coordinator and we do not have sanatizer so lets be sure...

Comment (ashtul) [reply coord/src/module.c:755]: Ran sanitizer locally and confirmed no memory leak

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**